### PR TITLE
DailyhuntBid Adapter: Add video support with Refactor/Optimizing

### PR DESCRIPTION
**Pull request by montu1996**
## Type of change

- [X] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [X] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
var adUnits = [
        {
            code: '/83414793/prebid_test_display',
            sizes: [[300, 250], [320, 50]],
            mediaTypes: {
                banner : {
                    sizes: [[300, 250], [320, 50]],
                }
            },
            bids: [
                {
                    bidder: 'dailyhunt',
                    params: {
                        placement_id: 1,
                        publisher_id: 1,
                        partner_name: 'dailyhunt',
                        device: {
                            ip: "182.23.143.212"
                        }
                    }
                }
            ]
        },
        {
            code: '/83414793/prebid_test_native',
            sizes: [[300, 250]],
            mediaTypes: {
                native: {
                    title: {
                        required: true
                    },
                    body: {
                        required: true
                    },
                    image: {
                        required: true
                    },
                    cta: {
                        required: true
                    }
                }
            },
            bids: [
                {
                    bidder: 'dailyhunt',
                    params: {
                        placement_id: 1,
                        publisher_id: 1,
                        partner_name: 'dailyhunt',
                        device: {
                            ip: "182.23.143.212"
                        }
                    }
                }
            ]
        },
        {
            code: '/83414793/prebid_test_video',
            mediaTypes: {
                video: {
                    playerSize: [1280, 720],
                    context: 'instream'
                }
            },
            bids: [
                {
                    bidder: 'dailyhunt',
                    params: {
                        placement_id: 1,
                        publisher_id: 1,
                        partner_name: 'dailyhunt',
                        device: {
                            ip: "182.23.143.212"
                        }
                        video: {
                            mimes: [
                                'video/mp4'
                            ]
                        }
                    }
                }
            ]
        }
    ];
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- eterno.newshunt@gmail.com
- [X] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by montu1996** [comment]
DOC PR: https://github.com/prebid/prebid.github.io/pull/1977

**Comment by montu1996** [comment]
Hello @bmwcmw ,

Any updates on this PR ?

**Comment by bmwcmw** [comment]
Hi @montu1996, I just begin reading the changes.
Could you split the PR into one for fix/cleanup and one for new features? I'll be appreciated.

**Comment by montu1996** [comment]
Hello @bmwcmw ,

We have tried keeping the cleanup and code refractor isolated as much as possible. as these code refractor and fix/cleanup are very tightly coupled with the new features , i dont see this is possible at this point of time. i would be happy to help you if there is any confusion or clarity . Looking forward for your kind coperation and advice.

**Comment by montu1996** [comment]
Hey @bmwcmw ,
Greetings from dailyhunt team.
Do you have any update for me to publish for our higher management team.
Apologies for the quick follow ups as this is a time sensitive and high revenue impacting feature addition for us so requesting for your kind help ?

File: modules/dailyhuntBidAdapter.js [hunk 1]
```diff
@@ -59,134 +341,55 @@ export const spec = {
 
   supportedMediaTypes: SUPPORTED_MEDIA_TYPES,
 
-  isBidRequestValid: bid => !!bid.params.partnerId,
+  isBidRequestValid: bid => !!bid.params.placement_id && !!bid.params.publisher_id && !!bid.params.partner_name,
```
Comment (bmwcmw) [inline modules/dailyhuntBidAdapter.js:344]: not sure if you also want to check validity of things like mediatype here, but it's not blocking

File: modules/dailyhuntBidAdapter.md [hunk 2]
```diff
@@ -55,15 +60,42 @@ Dailyhunt bid adapter supports Banner and Native.
...
+            bids: [
+                {
+                    bidder: 'dailyhunt',
+                    params: {
+                        placement_id: 1,
+                        publisher_id: 1,
+                        partner_name: 'dailyhunt',
+                        device: {
+                            ip: "182.23.143.212"
+                        }
```
[the remark this answers is not in the document]
Comment (montu1996) [reply modules/dailyhuntBidAdapter.md]: Hello @bmwcmw ,

Changes are fixed.

File: modules/dailyhuntBidAdapter.js (hunk 1 above)
Comment (montu1996) [reply modules/dailyhuntBidAdapter.js:344]: Hello @bmwcmw ,

Greetings from dailyhunt.
Thanks for reviewing. It is not required now.

**Pull request merged**